tests/unit/actor/test_request_list.py

```diff
@@ -0,0 +1,231 @@
+            'userData': {'some key': 'some value'},
+            'headers': {'h1': 'v1', 'h2': 'v2'},
+        },
+    ],
+    ids=['minimal', 'all_options'],
+)
+async def test_request_list_open_request_types(request_method: HttpMethod, optional_input: dict[str, Any]) -> None:
+    """Test proper request list generation from both minimal and full inputs for all method types for simple input."""
+    minimal_request_dict_input = {
+        'url': 'https://www.abc.com',
+        'method': request_method,
+    }
+    request_dict_input = {**minimal_request_dict_input, **optional_input}
+
+    request_list = await RequestList.open(actor_start_urls_input=[request_dict_input])
+    assert not await request_list.is_empty()
+    request = await request_list.fetch_next_request()
+    assert request is not None
+    assert await request_list.is_empty()
+
+    assert request.method == request_dict_input['method']
+    assert request.url == request_dict_input['url']
+    assert request.payload == request_dict_input.get('payload', '').encode('utf-8')
+    expected_user_data = UserData()
+    if 'userData' in optional_input:
+        for key, value in optional_input['userData'].items():
+            expected_user_data[key] = value
+    assert request.user_data == expected_user_data
+    assert request.headers.root == optional_input.get('headers', {})
+
+
+def _create_dummy_response(read_output: Iterator[str]) -> HttpResponse:
+    """Create dummy_response that will iterate through read_output when called like dummy_response.read()"""
+
+    class DummyResponse(HttpResponse):
+        @property
+        def http_version(self) -> str:
+            return ''
+
+        @property
+        def status_code(self) -> int:
+            return 200
+
+        @property
+        def headers(self) -> HttpHeaders:
+            return HttpHeaders()
+
+        def read(self) -> bytes:
+            return next(read_output).encode('utf-8')
+
+    return DummyResponse()
+
+
+async def test__request_list_open_from_url_correctly_send_requests() -> None:
+    """Test that injected HttpClient's method send_request is called with properly passed arguments."""
+    actor_start_urls_input: list[dict[str, Any]] = [
+        {
+            'requestsFromUrl': 'https://abc.dev/file.txt',
+            'method': 'GET',
+        },
+        {
+            'requestsFromUrl': 'https://www.abc.dev/file2',
+            'method': 'PUT',
+        },
+        {
+            'requestsFromUrl': 'https://www.something.som',
+            'method': 'POST',
+            'headers': {'key': 'value'},
+            'payload': 'some_payload',
+            'userData': {'another_key': 'another_value'},
+        },
+    ]
+
+    mocked_read_outputs = ('' for url in actor_start_urls_input)
+    http_client = HttpxHttpClient()
+    with mock.patch.object(
+        http_client, 'send_request', return_value=_create_dummy_response(mocked_read_outputs)
+    ) as mocked_send_request:
```

Comment:
Done

Reply:
If you want to go with mocking, then please mock `BaseHttpClient` instead of a concrete implementation. I have a feeling that using `respx` could result in a more readable code - did you try it?

Reply:
Yes, respx is actually quite convenient to use. Thanks for the tip.